developer-workflow/c-api.rst

```diff
@@ -134,6 +145,101 @@ When moving existing tests, feel free to replace ``TestError`` with
 ``PyExc_AssertionError`` unless actually testing custom exceptions.
 
 
+.. _unstable-capi:
+
+Unstable C API
+==============
+
+The unstable C API tier is meant for extensions that need tight integration
+with the interpreter, like debuggers and JIT compilers.
+Users of this tier may need to change their code with every minor release.
+
+In many ways, this tier is like the general C API:
+
+- it's available when ``Python.h`` is included normally,
+- it should be defined  in ``Include/cpython/``,
+- it requires tests, so we don't break it unintentionally
+- it requires docs, so both we and the users
+  can agree on the expected behavior,
+- it is tested and documented in the same way.
+
+The differences are:
+
+- Names of functions structs, macros, etc. start with the ``PyUnstable_``
+  prefix. This defines what's in the unstable tier.
+- The unstable API can change in minor versions, without any deprecation
+  period.
+- A stability note is appears in the docs.
+  This happens automatically, based on the name
+  (via :cpy-file:`Doc/tools/extensions/c_annotations.py`).
+
+Despite being “unstable”, there are rules to make sure third-party code can
+use this API reliably:
+
+* Changes and removals can be done in minor releases (3.x.0, including
+  Alphas and Betas for 3.x.0).
+* Adding new unstable API *for an existing feature* is allowed even after
+  Beta feature freeze, up until the first Release Candidate.
+  Consensus on Core Development Discourse or is needed in the Beta period.
+* Backwards-incompatible changes should make existing C callers fail to compile.
+  For example, arguments should be added/removed, or a function should be
+  renamed.
+* When moving API into or out of the Unstable tier, the old name
+  should continue to be available (but deprecated) until an incompatible
+  change is made. In other words, while we're allowed to break calling code,
+  we shouldn't break it *unnecessarily*.
+
+
+Moving API from the public tier to unstable
+-------------------------------------------
```

Comment:
This either needs an article, or a plural (`s`). Since the section is framed in terms of a single API, I've suggested the former:

```suggestion
Moving an API from the public tier to unstable
----------------------------------------------
```

Reply:
Other occurrences of "Unstable" are capitalized, so maybe this should be too.